storage/aws/efs/main.tf

```diff
@@ -1,26 +1,40 @@
+# Availability zones
+data "aws_subnet" "private_subnet" {
+  for_each   = toset(var.vpc_cidr_block_private)
+  cidr_block = each.key
+
+  # dummy ternary impose dependency on subnets even though depends_on does not work here
+  vpc_id = var.vpc_subnet_ids != null ? var.vpc_id : var.vpc_id
+}
```

Comment:
I think it is much better to give the subnets information directly as input of the module. Re-reading a resource that has been created in the same deployment can lead to strange behavior like some information not propagated upon infrastructure change.